sklearn/cluster/tests/test_optics.py

```diff
@@ -29,14 +32,122 @@
 X = np.vstack((C1, C2, C3, C4, C5, C6))
 
 
+@pytest.mark.parametrize(
+    ('r_plot', 'end', 'index'),
+    [[[10, 8.9, 8.8, 8.7, 7, 10], 3, 4],
+     [[10, 8.9, 8.8, 8.7, 8.6, 7, 10], 0, 4],
+     [[10, 8.9, 8.8, 8.7, 7, 6, np.inf], 4, 5],
+     [[10, 8.9, 8.8, 8.7, 7, 6, np.inf], 4, 5],
+     ])
+def test_extend_downward(r_plot, end, index):
```

Comment:
I don't think we need these two tests (save some CI resources), feel free keep them if you like.

Reply:
they're fast, and I've fixed and found bugs using them, so I'd say they do no harm and we better keep them.